third_party/terraform/utils/iam_test.go

```diff
@@ -0,0 +1,1113 @@
+				},
+				{
+					Role:    "role-2",
+					Members: []string{"member-1", "member-2"},
+				},
+				{
+					Role:    "role-3",
+					Members: []string{"member-1"},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		got := mergeBindings(tc.input)
+		if !compareBindings(got, tc.expect) {
+			t.Errorf("Unexpected value for mergeBindings(%s).\nActual: %s\nExpected: %s\n",
+				debugPrintBindings(tc.input), debugPrintBindings(got), debugPrintBindings(tc.expect))
+		}
+	}
+}
+
+func TestIamRemoveAllBindingsWithRole(t *testing.T) {
+	testCases := []struct {
+		input  []*cloudresourcemanager.Binding
+		role   string
+		expect []*cloudresourcemanager.Binding
+	}{
+		// No-op
+		{
+			input:  []*cloudresourcemanager.Binding{},
+			role:   "role-1",
+			expect: []*cloudresourcemanager.Binding{},
+		},
+		// Remove one binding
+		{
+			input: []*cloudresourcemanager.Binding{
+				{
+					Role:    "role-1",
+					Members: []string{"member-1", "member-2"},
+				},
+			},
+			role:   "role-1",
+			expect: []*cloudresourcemanager.Binding{},
+		},
+		// Remove multiple bindings
+		{
+			input: []*cloudresourcemanager.Binding{
+				{
+					Role:    "role-1",
+					Members: []string{"member-1", "member-2"},
+				},
+				{
+					Role:    "role-1",
+					Members: []string{"member-3"},
+				},
+			},
+			role:   "role-1",
+			expect: []*cloudresourcemanager.Binding{},
+		},
+		// Remove multiple bindings and leave some.
+		{
+			input: []*cloudresourcemanager.Binding{
+				{
+					Role:    "role-1",
+					Members: []string{"member-1", "member-2"},
+				},
+				{
+					Role:    "role-2",
+					Members: []string{"member-1"},
+				},
+				{
+					Role:    "role-3",
+					Members: []string{"member-1", "member-3"},
+				},
+				{
+					Role:    "role-1",
+					Members: []string{"member-2"},
+				},
+				{
+					Role:    "role-2",
+					Members: []string{"member-1", "member-2"},
+				},
+			},
+			role: "role-1",
+			expect: []*cloudresourcemanager.Binding{
+				{
+					Role:    "role-2",
+					Members: []string{"member-1"},
+				},
+				{
+					Role:    "role-3",
+					Members: []string{"member-1", "member-3"},
+				},
+				{
+					Role:    "role-2",
+					Members: []string{"member-1", "member-2"},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		got := removeAllBindingsWithRole(tc.input, tc.role)
+		if !compareBindings(got, tc.expect) {
+			t.Errorf("Got unexpected value for removeAllBindingsWithRole(%s, %s).\nActual: %s\nExpected: %s",
+				debugPrintBindings(tc.input), tc.role, debugPrintBindings(got), debugPrintBindings(tc.expect))
+		}
+	}
+}
+
+func TestIamSubtractFromBindings(t *testing.T) {
+	testCases := []struct {
+		input  []*cloudresourcemanager.Binding
+		remove []*cloudresourcemanager.Binding
+		expect []*cloudresourcemanager.Binding
+	}{
+		{
+			input:  []*cloudresourcemanager.Binding{},
+			remove: []*cloudresourcemanager.Binding{},
+			expect: []*cloudresourcemanager.Binding{},
+		},
+		// Empty expect should no-op return empty
```

Comment:
nit: s/expect/input for most of these comments

Reply:
intellij renaming is wild